### PR TITLE
Pin libmxml to 3.3.1 in macOS CI

### PR DESCRIPTION
**Pull request by stevedlawrence**
Homewbrew upated to 4.x and currently fails CI

DAFFODIL-2889

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -102,7 +102,10 @@ jobs:
 
       - name: Install Dependencies (macOS)
         if: runner.os == 'macOS'
-        run: brew install libmxml
+        run: |
+          brew tap-new local/libmxml
+          brew extract --version 3.3.1 libmxml local/libmxml
+          brew install libmxml@3.3.1
```
Comment (tuxji) [inline .github/workflows/main.yml:108]: I'm curious why the first two commands are needed.  Are they always required even though the page I saw didn't mention them, or is there something about the GitHub runner which makes them necessary?  I have not set up a programming environment on macOS so I don't know much about brew.

**Review by tuxji — APPROVED** [review]
+1

File: .github/workflows/main.yml (hunk 1 above)
Comment (stevedlawrence) [reply .github/workflows/main.yml:108]: Yeah, I found that the normal @xyz thing only works if the homebrew-core repo includes multiple versions. Homebrew only seems to do this for popular packages where people might need multiple versions, like node.js. For less popular packages, like libmxml, they only support the latest version. Seems to be a pretty basic package manager.

So essentially what this does is it creates a local repo (`brew tap-new`), finds and extracts 3.3.1 out of the git history of homebrew-core into that local repo (`brew extract`), and then installs that local version.

This is the only way I could find to do it. It seems kindof developer-y/experimental, so it feels like it we might want it to just be a short term solution. In the long term, we probably need to figure out why 4.x doesn't work and see if we can support that. But this at least allows our CI to pass.

Comment (stevedlawrence) [reply .github/workflows/main.yml:108]: Note, I also don't know much about macOS and homebrew. This was all just based on googling and lots of trial and error until I found something that worked. There might be a better way to do it.

Comment (pkatlic) [reply .github/workflows/main.yml:108]: It does seem that this is the recommended approach to specifying an older version due to the way that homebrew is managed.

**Pull request merged**